crates/sui-snapshot/src/lib.rs

```diff
@@ -292,3 +318,29 @@ pub fn create_file_metadata(
     };
     Ok(file_metadata)
 }
+
+pub async fn setup_db_state(
+    epoch: u64,
+    perpetual_db: Arc<AuthorityPerpetualTables>,
+    checkpoint_store: Arc<CheckpointStore>,
+    committee_store: Arc<CommitteeStore>,
+) -> Result<()> {
+    // This function should be called once state accumulator based hash verification
+    // is complete and live object set state is downloaded to local store
+    let system_state_object = get_sui_system_state(&perpetual_db)?;
+    let new_epoch_start_state = system_state_object.into_epoch_start_state();
+    let next_epoch_committee = new_epoch_start_state.get_sui_committee();
+    let last_checkpoint = checkpoint_store
+        .get_epoch_last_checkpoint(epoch)
+        .expect("Error loading last checkpoint for current epoch")
+        .expect("Could not load last checkpoint for current epoch");
+    let epoch_start_configuration =
+        EpochStartConfiguration::new(new_epoch_start_state, *last_checkpoint.digest());
+    perpetual_db
+        .set_epoch_start_configuration(&epoch_start_configuration)
+        .await?;
+    committee_store.insert_new_committee(&next_epoch_committee)?;
+    checkpoint_store.update_highest_synced_checkpoint(&last_checkpoint)?;
+    checkpoint_store.update_highest_executed_checkpoint(&last_checkpoint)?;
```

Comment:
If we update highest synced to the last checkpoint, we are also advertising via state sync that we have all those full checkpoint contents, which we don't. I don't think we can do this safely until we have the lowest synced watermark.

Reply:
This makes sense, I've fixed the code and removed `update_highest_synced_checkpoint` from the list.